runners/core-java/src/main/java/org/apache/beam/runners/core/SimpleDoFnRunner.java

```diff
@@ -172,9 +175,10 @@ public void onTimer(
   }
 
   private void invokeProcessElement(WindowedValue<InputT> elem) {
+    DoFnProcessContext context = new DoFnProcessContext(elem);
     // This can contain user code. Wrap it in case it throws an exception.
-    try {
-      invoker.invokeProcessElement(new DoFnProcessContext(elem));
+    try (ScopedSideInputContext ignored = setCurrentSideInputContext(context)) {
```

Comment:
<!--new_thread; commit:2b7b6939227e3e105feee6eddf1d6c943c40815a; resolved:0-->
nit: Technically not ignored since your relying on the try with resources to do work for you.